plugins/inputs/intel_powerstat/services/msr.go

```
@@ -0,0 +1,297 @@
+	if err != nil {
+		return err
+	}
+
+	for _, cpuPath := range cpuPaths {
+		core := strings.TrimPrefix(filepath.Base(cpuPath), cpuPrefix)
+		m.cpuCoresData[core] = &data.MsrData{
+			Mperf:        0,
+			Aperf:        0,
+			Tsc:          0,
+			C3:           0,
+			C6:           0,
+			C7:           0,
+			ThrottleTemp: 0,
+			Temp:         0,
+			MperfDelta:   0,
+			AperfDelta:   0,
+			TscDelta:     0,
+			C3Delta:      0,
+			C6Delta:      0,
+			C7Delta:      0,
+		}
+	}
+
+	return nil
+}
+
+func (m *MsrServiceImpl) getMsrOffsets() map[string]int64 {
+	return m.msrOffsets
+}
+
+func (m *MsrServiceImpl) calculateMsrOffsets() {
+	m.msrOffsets = map[string]int64{
+		c3StateResidencyLocation:           m.parseStringHexToInt(c3StateResidencyLocation),
+		c6StateResidencyLocation:           m.parseStringHexToInt(c6StateResidencyLocation),
+		c7StateResidencyLocation:           m.parseStringHexToInt(c7StateResidencyLocation),
+		maximumFrequencyClockCountLocation: m.parseStringHexToInt(maximumFrequencyClockCountLocation),
+		actualFrequencyClockCountLocation:  m.parseStringHexToInt(actualFrequencyClockCountLocation),
+		TimestampCounterLocation:           m.parseStringHexToInt(TimestampCounterLocation),
+		throttleTemperatureLocation:        m.parseStringHexToInt(throttleTemperatureLocation),
+		temperatureLocation:                m.parseStringHexToInt(temperatureLocation),
+	}
+}
+
+func (m *MsrServiceImpl) parseStringHexToInt(s string) int64 {
+	parsedInt, err := strconv.ParseInt(s, 0, 64)
+	if err != nil {
+		m.log.Errorf("error on parsing offset %s, err: %v", s, err)
+		return 0
+	}
+
+	return parsedInt
+}
+
+// NewMsrServiceWithFs returns new RaplServiceImpl struct with given FileService.
+func NewMsrServiceWithFs(logger telegraf.Logger, fs FileService) *MsrServiceImpl {
+	msrService := &MsrServiceImpl{
+		fs:  fs,
+		log: logger,
+	}
+	err := msrService.setCPUCores()
```

The only location where this call can fail is if the well-formatted patterns passed to `filepath.Glob()` are broken. We then have a much deeper problem. So I'd rather return an error here.

---

I understand, this is almost impossible to happen that this pattern is not well-formatted but for security reasons and for the sake of handling every error we would like to leave it as it is.